assets/javascripts/file_browser.js

```diff
@@ -6,8 +6,28 @@ window.addEventListener("load", function() {
 			var modal = document.createElement("div");
 			modal.setAttribute("id", "file-browser");
 
+			function cancelOnEsc(e) {
+				document.removeEventListener("keyup", cancelOnEsc);
+
+				if(e.key === "Escape") {
+					reject("cancel");
+					modal.parentElement.removeChild(modal);
+				}
+			}
+
+			modal.addEventListener("click", function() {
+				document.removeEventListener("keyup", cancelOnEsc);
+				reject("cancel");
+				modal.parentElement.removeChild(modal);
+			});
+
+			document.addEventListener("keyup", cancelOnEsc);
+
 			var browser = document.createElement("div");
 			browser.className = "contents";
+			browser.addEventListener("click", function(e) {
+				e.stopPropagation();
```

Comment:
Why do we stop propagation? Is this so that it doesn't trigger the click event on modal? If so, can we just add a check in the modal click event that the target is not contents instead? I've had issues in the past where stopPropagation has caused debugging issues because you try to add a handler somewhere else and don't realize that somebody called stopPropagation.